pkg/dataobj/sections/pointers/reader_test.go

```diff
@@ -0,0 +1,681 @@
+		{Path: "path3", Section: 3, PointerKind: pointers.PointerKindStreamIndex, StreamID: 30, StreamIDRef: 300, StartTs: unixTime(50), EndTs: unixTime(60), LineCount: 15, UncompressedSize: 3072},
+	})
+
+	var (
+		pathCol     = sec.Columns()[0]
+		sectionCol  = sec.Columns()[1]
+		streamIDCol = sec.Columns()[3]
+	)
+
+	r := pointers.NewReader(pointers.ReaderOptions{
+		Columns:   []*pointers.Column{pathCol, sectionCol, streamIDCol},
+		Allocator: memory.DefaultAllocator,
+		Predicates: []pointers.Predicate{
+			pointers.LessThanPredicate{
+				Column: streamIDCol,
+				Value:  scalar.NewInt64Scalar(25),
+			},
+		},
+	})
+
+	actualTable, err := readTable(context.Background(), r)
+	require.NoError(t, err)
+
+	actual, err := arrowtest.TableRows(memory.DefaultAllocator, actualTable)
+	require.NoError(t, err)
+
+	expected := arrowtest.Rows{
+		{"path.path.utf8": "path1", "section.int64": int64(1), "stream_id.int64": int64(10)},
+		{"path.path.utf8": "path2", "section.int64": int64(2), "stream_id.int64": int64(20)},
+	}
+	require.Equal(t, expected, actual)
+}
+
+// TestReaderWithTimestampPredicates tests reading with timestamp predicates.
+func TestReaderWithTimestampPredicates(t *testing.T) {
+	sec := buildSection(t, []pointers.SectionPointer{
+		{Path: "path1", Section: 1, PointerKind: pointers.PointerKindStreamIndex, StreamID: 10, StreamIDRef: 100, StartTs: unixTime(10), EndTs: unixTime(20), LineCount: 5, UncompressedSize: 1024},
+		{Path: "path2", Section: 2, PointerKind: pointers.PointerKindStreamIndex, StreamID: 20, StreamIDRef: 200, StartTs: unixTime(30), EndTs: unixTime(40), LineCount: 10, UncompressedSize: 2048},
+		{Path: "path3", Section: 3, PointerKind: pointers.PointerKindStreamIndex, StreamID: 30, StreamIDRef: 300, StartTs: unixTime(50), EndTs: unixTime(60), LineCount: 15, UncompressedSize: 3072},
+	})
+
+	var (
+		pathCol         = sec.Columns()[0]
+		sectionCol      = sec.Columns()[1]
+		minTimestampCol = sec.Columns()[5]
+		maxTimestampCol = sec.Columns()[6]
+	)
+
+	r := pointers.NewReader(pointers.ReaderOptions{
+		Columns:   []*pointers.Column{pathCol, sectionCol, minTimestampCol, maxTimestampCol},
+		Allocator: memory.DefaultAllocator,
+		Predicates: []pointers.Predicate{
+			pointers.GreaterThanPredicate{
+				Column: minTimestampCol,
+				Value:  scalar.NewTimestampScalar(arrow.Timestamp(unixTime(25).UnixNano()), &arrow.TimestampType{Unit: arrow.Nanosecond}),
+			},
+			pointers.LessThanPredicate{
+				Column: maxTimestampCol,
+				Value:  scalar.NewTimestampScalar(arrow.Timestamp(unixTime(55).UnixNano()), &arrow.TimestampType{Unit: arrow.Nanosecond}),
+			},
+		},
+	})
+
+	actualTable, err := readTable(context.Background(), r)
+	require.NoError(t, err)
+
+	actual, err := arrowtest.TableRows(memory.DefaultAllocator, actualTable)
+	require.NoError(t, err)
+
+	expected := arrowtest.Rows{
```

Comment:
I don't think this is correct: A range of 25-55 should cover both path2 and path3 rows. A timestamp range should match any stream it overlaps with, it doesn't have to completely cover it

Reply:
Updated